src/test/java/bit/javaoop/people/FixedBudgetManagerTest.java

```diff
@@ -0,0 +1,96 @@
+package bit.javaoop.people;
+
+import bit.javaoop.CompanyAppException;
+import bit.javaoop.CompanyAppExceptionMessage;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class FixedBudgetManagerTest {
+
+    private Manager manager;
+    private AbstractEmployee employee;
+    private double testHiringBudget;
+
+    @BeforeEach
+    void init() {
+        testHiringBudget = 30_000;
+        manager = new FixedBudgetManager("Manager", 30_000, testHiringBudget);
+        employee = new Employee("E1", 1_000);
+    }
+
+    @Test
+    void testIfCanNotHireEmployeeWithTooBigSalary() throws CompanyAppException {
+        employee = new Employee("E1", 100_000);
+
+        assertFalse(manager.canHireEmployee(employee));
+    }
+
+    @Test
+    void testIfCanHireEmployeeWithNormalSalary() throws CompanyAppException {
+        assertTrue(manager.canHireEmployee(employee));
+    }
+
+    @Test
+    void testIfCanHireSecondEmployeesUnderBudget() throws CompanyAppException {
+        AbstractEmployee employee2 = new Employee("E2", 3_000);
+        manager.hire(employee);
+
+        assertTrue(manager.canHireEmployee(employee2));
+    }
+
+    @Test
+    void testIfCanNotHireSecondEmployeesAboveBudget() throws CompanyAppException {
+        AbstractEmployee employee2 = new Employee("E2", 30_000);
+        manager.hire(employee);
+
+        assertFalse(manager.canHireEmployee(employee2));
+    }
+
+    @Test
+    void testHiringAboveBudget() {
+        employee = new Employee("E1", 100_000);
+
+        Exception ex = assertThrows(CompanyAppException.class, () -> manager.hire(employee));
+        Assertions.assertEquals(CompanyAppExceptionMessage.BUDGET_OVERFLOW.toString(), ex.getMessage());
+    }
```

Comment:
Nieźle napisany test, skąd się dowiedziałeś o lambdach? No chyba, że zrobiłeś copy-paste'a ze stack'a xd

Reply:
Jeśli chodzi o lambdy to w zasadzie Java 8 jest już jakieś 3-4 lata więc wypada coś wiedzieć ;) 
Co do samego rozwiązania to assertThrows() akurat nie znałem (jeśli się nie mylę dodane w JUnit5 bo wcześniej tego nie widziałem) i tutaj Stack jak najbardziej pomógł :)